extmod/modtls_mbedtls.c

```diff
@@ -100,6 +100,12 @@ static void mbedtls_debug(void *ctx, int level, const char *file, int line, cons
 }
 #endif
 
+#if defined(MBEDTLS_PEM_PARSE_C)
+static int mbedtls_is_pem(const byte *data, size_t len) {
+    return (len >= 10) && (memcmp(data, "-----BEGIN", 10) == 0);
```

Comment:
> Since all PEM data starts with "-----BEGIN" this is used to check if the data is PEM.

This is frustratingly not quite true, because PEM is an outgrowth of a mail format it can contain preceding plaintext.

Suggest using `strstr()` like mbedTLS does here: https://github.com/Mbed-TLS/mbedtls/blob/cfe8c4c57d7e71ea4763252e76b20fe019040eb8/tf-psa-crypto/drivers/builtin/src/pkparse.c#L1337

Reply:
It's relatively minor, but I'd say if there's a PEM certificate file that is accepted by desktop mbedTLS and openSSL then it should be accepted by MicroPython as well. Which means allowing preceding text, even if it's not something commonly used.

What do you think about combining my commit and your test addition for this? My version of the fix already uses strstr and it's a little more factored (less duplicate code), and they both do more or less the same thing.

Reply:
@peterzuger The change https://github.com/micropython/micropython/commit/b180f1fcb1af54505a170f61a3d08645f2ccd1bf that @projectgus refers to is a little cleaner than the changes here, because it factors the extraction of cert data into a helper function, and only has one location with a conditional check on `MBEDTLS_PEM_PARSE_C`.  So I would prefer that simpler commit.

The test from this PR is very good though, we definitely want that.

What do you think, shall we take @projectgus 's commit and your test?

Reply:
I updated this to use `strstr`.